pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -712,6 +715,37 @@ public String getTableState(
     }
   }
 
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Authenticate(AccessType.UPDATE)
+  @Path("/rebalanceStatus/{jobId}")
+  @ApiOperation(value = "Gets detailed stats of a rebalance operation",
+      notes = "Gets detailed stats of a rebalance operation")
+  public ServerRebalanceJobStatusResponse rebalanceStatus(
+      @ApiParam(value = "Rebalance Job Id", required = true) @PathParam("jobId") String jobId)
+      throws JsonProcessingException {
+    Map<String, String> controllerJobZKMetadata =
+        _pinotHelixResourceManager.getControllerJobZKMetadataForJobType(jobId, ControllerJobType.TABLE_REBALANCE);
+
+    if (controllerJobZKMetadata == null) {
+      throw new ControllerApplicationException(LOGGER, "Failed to find controller job id: " + jobId,
+          Response.Status.NOT_FOUND);
+    }
+    TableRebalanceProgressStats tableRebalanceProgressStats =
+        JsonUtils.stringToObject(controllerJobZKMetadata.get(RebalanceConfigConstants.REBALANCE_PROGRESS_STATS),
+            TableRebalanceProgressStats.class);
+    Long timeSinceStartInSecs = 0L;
+    if (!tableRebalanceProgressStats.getStatus().toString().equals(RebalanceResult.Status.DONE.toString())) {
+      timeSinceStartInSecs =
+          (long) (System.currentTimeMillis() - tableRebalanceProgressStats.getStartTimeInMilliseconds()) / 1000;
```

Comment:
(minor) The case is redundant
```suggestion
          (System.currentTimeMillis() - tableRebalanceProgressStats.getStartTimeInMilliseconds()) / 1000;
```

Reply:
Its needed, as return type is double. Long (in seconds) makes sense for reporting stats